Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -1460,42 +1460,76 @@ bool CStaticFunctionDefinitions::SetElementModel(CClientEntity& Entity, unsigned
         {
             // Grab the model
             CClientPed& Ped = static_cast<CClientPed&>(Entity);
-            if (Ped.GetModel() == usModel)
+            const unsigned short usCurrentModel = Ped.GetModel();
+
+            if (usCurrentModel == usModel)
                 return false;
-            if (!CClientPlayerManager::IsValidModel(usModel))
+
+            if (!Ped.SetModel(usModel))

```

Comment:
I'm guessing this checks IsValidModel internally?

Reply:
https://github.com/multitheftauto/mtasa-blue/blob/14a5c3755b14512819ecd4bdb89f3161d79316ca/Client/mods/deathmatch/logic/CClientPed.cpp#L1003-L1006